Lib/test/test_peg_generator/test_c_parser.py

```diff
@@ -402,3 +402,34 @@ def test_error_in_rules(self) -> None:
             parse.parse_string("a", mode=0)
         """
         self.run_test(grammar_source, test_source)
+
+    def test_no_soft_keywords(self) -> None:
+        grammar_source = """
+        start: expr+ NEWLINE? ENDMARKER
+        expr: 'foo'
+        """
+        grammar = parse_string(grammar_source, GrammarParser)
+        parser_source = generate_c_parser_source(grammar)
+        assert "expect_soft_keyword" not in parser_source
+
+    def test_soft_keywords(self) -> None:
+        grammar_source = """
+        start: expr+ NEWLINE? ENDMARKER
+        expr: "foo"
+        """
+        grammar = parse_string(grammar_source, GrammarParser)
+        parser_source = generate_c_parser_source(grammar)
+        assert "expect_soft_keyword" in parser_source
+
+    def test_soft_keywords_parse(self) -> None:
+        grammar_source = """
+        @subheader "#undef D\\n#define D(x) x\\n"
```

Comment:
Question: Why Is this needed? In the latest version this macro expands to nothing outside the interpreter core:

https://github.com/python/cpython/blob/master/Tools/peg_generator/pegen/c_generator.py#L32

Reply:
Whoops, that was a failed attempt at debugging. Deleting.